src/Relation/Nullary/Decidable.agda

```diff
@@ -60,8 +60,8 @@ module _ {a₁ a₂ b₁ b₂} {A : Setoid a₁ a₂} {B : Setoid b₁ b₂}
 -- A lemma relating True and Dec
 
 True-↔ : (dec : Dec P) → Irrelevant P → True dec ↔ P
-True-↔ (true  because  [p]) irr = mk↔′ (λ _ → invert [p]) _ (irr (invert [p])) cong′
-True-↔ (false because ofⁿ ¬p) _ = mk↔′ (λ ()) (invert (ofⁿ ¬p)) (⊥-elim ∘ ¬p) λ ()
+True-↔ (yes p) irr = mk↔′ (λ _ → p) _ (irr p) cong′
```

Comment:
is this definition now too strict?

I confess I find the use of `Reflects.invert` hard to understand, because it only ever seem to occur in the scope of pattern-matching context, of the form `true because ...` or else `false because ...` in which the relevant clause of `invert` can be statically known...

In other words, if we were instead to define
```agda
pattern lazy-yes [p] = true because [p]
pattern lazy-no [¬p] = false because [¬p]
```
then is there any distinction between pattern-matches of the form
```agda
f (lazy-yes [p]) = ... invert [p] ...
f (lazy-no [¬p] = ... invert [¬p] ...
```
and the corresponding
```agda
f (yes p) = ... p ...
f (no ¬p = ... ¬p ...
```
Apologies if this is 'obvious', but I'm confused...

Reply:
Well, I've now restored (a version of) the less strict definition, so resolving this now.